app/src/main/java/com/jerboa/model/SiteViewModel.kt

```diff
@@ -211,10 +216,22 @@ class SiteViewModel(private val accountRepository: AccountRepository) : ViewMode
         }
     }
 
-    fun showScores(): Boolean {
+    // TODO this should probably be persisted rather than waited for
+    // For the current default, just use FullScores
+    fun voteDisplayMode(): VoteDisplayMode {
+        val defaultMode = VoteDisplayMode.Full
```

Comment:
Here's what makes this all temporary. When this actually comes back from the siteRes, then we can remove it.

Reply:
If we don't persist this? Don't we keep resetting it each time the user opens the app again?

Reply:
The question is : Which local user settings should we be saving to our local database?

I think we only really need to save things required *before* the first fetch, so `default_sort_type` . But as for other things that come back with `getSite` ? I'm not sure.

Reply:
We should discuss this, but not only the context of this setting but like sort options on each screen too.

For backwards compatibility it seems that we will need to save the above setting too unless it not persistenting for older versions is intended.

Also I don't think it is guaranteed that getSite has returned before feed loads? So initially it won't be correct occasionally?

Reply:
> Also I don't think it is guaranteed that getSite has returned before feed loads? So initially it won't be correct occasionally?

Yep, it won't be correct until after GetSite has loaded. In practice I haven't seen a case where the any feed loads *faster* than GetSite tho. 

In that `SiteViewModel`, the current things that *could* be persisted are admins, followList, voteDisplayMode, enableDownvotes, and showAvatars.

If we decide to persist them, we should do that as a later PR, all at once tho. I'm not 100% convinced its worth it... but we do already have everything set up (persisting them would work exactly like `default_sort_type`)

Reply:
> Yep, it won't be correct until after GetSite has loaded. In practice I haven't seen a case where the any feed loads faster than GetSite tho.

That must be very situational. I just tested 5 times on my phone and all 5 times it did the feed request before the site request. You can check this by checking the log order for the initiation request

![image](https://github.com/dessalines/jerboa/assets/67873169/1c66f34c-e3a6-4987-8238-740bf1dde8b1)


> admins, followList, voteDisplayMode, enableDownvotes, and showAvatars.

I care less about those are more about things we don't persist at all, like the sort types on each screen (communityview, postview, ...)

Maybe we should devise a much more flexible system that doesn't require so much overhead (as code) for persisting more fields.

Reply:
Sort types on each screen should just use the `default_sort_type` , which is already saved in the DB.

We *could* add a string / JSON field for user settings, but I really don't like putting untyped things in the DB, as that makes migrations and many other things a nightmare.

Reply:
We can't reuse default sort because it is in a different context+ different enum (sometimes)

For example comment sort options
or sort options in saved/own profile would be different context where you wouldn't want to reuse the same sort as for the feed

Reply:
I'll add default_comment_sort for users and sites eventually: https://github.com/LemmyNet/lemmy/issues/4128

I'm not sure what would be best, but at least for now lets just add more columns to the account table as necessary.